src/main/java/uk/gov/justice/probation/courtcasematcher/event/EventListener.java

```diff
@@ -65,9 +63,16 @@ public void courtCaseEvent(CourtCaseSuccessEvent courtCaseEvent) {
     @AllowConcurrentEvents
     @Subscribe
     public void courtCaseUpdateEvent(CourtCaseUpdateEvent courtCaseEvent) {
-        CourtCase courtCase = courtCaseEvent.getCourtCase();
-        log.info("Upsert case no {} for court {}", courtCase.getCaseNo(), courtCase.getCourtCode());
-        courtCaseService.saveCourtCase(courtCaseEvent.getCourtCase());
+        final CourtCase courtCase = courtCaseEvent.getCourtCase();
+        log.info("Upsert case no {} with crn {} for court {}", courtCase.getCaseNo(), courtCase.getCrn(), courtCase.getCourtCode());
+        Optional.ofNullable(courtCase.getCrn())
+            .ifPresentOrElse(crn -> {
+                log.debug("refresh probation status for case no {} with crn {} for court {}", courtCase.getCaseNo(), crn, courtCase.getCourtCode());
+                courtCaseService.updateProbationStatusDetail(courtCase)
+                    .doOnError(throwable -> courtCaseService.saveCourtCase(courtCase))
```

Comment:
Do we need to log an error here? What happens if the probationStatus endpoint returns a 404? Presumably that will return empty so no court-case will be saved?

Reply:
The error is logged in getProbationStatusDetail in CourtCaseRestClient which always returns an empty Mono so there'll be a log when it happens. 
`GET failed for retrieving the offender probation status detail for path {}. Will return empty status`

The idea was that whatever happens in that call, it's not a fatal error, we should carry on and save what we have.

Actually, I am now thinking that we won't actually have an error there. The rest client catches the error, logs it, returns an empty Mono. `CourtCaseService.updateProbationStatusDetail` returns the original court case if it receives an empty Mono. So the caller here in EventListener will only ever get an updated court case or the original one.